### PR TITLE
feat(hypershift): IBM Cloud E2E CI enablement

### PR DESCRIPTION
**Pull request by hasueki**
Connects to https://github.com/openshift/hypershift/issues/652

This enables IBM Cloud CI for https://github.com/openshift/hypershift/tree/main.

### CODE REVIEW

**Review by csrwng — COMMENTED** [review]
Looks good, some comments.

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml [hunk 1]
```diff
@@ -120,6 +120,24 @@ tests:
       GATHER_CAPI_LOGS: "true"
       TEST_SUITE: minimal
     workflow: assisted-baremetal
+- as: e2e-ibmcloud-iks
```
[the remark this answers is not in the document]
Comment (hasueki) [reply ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:129]: Added

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml [hunk 2]
```diff
@@ -120,6 +120,24 @@ tests:
...
+  skip_if_only_changed: (^(docs|examples|enhancements)/)|(^[A-Z]+\.md$)|((^|/)OWNERS$)
+  steps:
+    env:
+      BOM_VERSION: 4.10.26_openshift
+      MANAGED_CLUSTER_TYPE: iks
+      MANAGEMENT_CLUSTER_KUBECONFIG: /etc/ibmcloud/secrets/iks.kubeconfig
+      VERIFY_COMPLIANCE_OPERATOR_SCAN: "false"
+    workflow: hypershift-ibmcloud-e2e
+- as: e2e-ibmcloud-roks
```
[the remark this answers is not in the document]
Comment (hasueki) [reply ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:139]: Added

File: ci-operator/step-registry/hypershift/ibmcloud/OWNERS [hunk 3]
```diff
@@ -0,0 +1,11 @@
+approvers:
```
[the remark this answers is not in the document]
Comment (hasueki) [reply ci-operator/step-registry/hypershift/ibmcloud/OWNERS:1]: Gotcha, added

**Comment by hasueki** [comment]
Thanks for the review @csrwng! Updated the PR, PTAL.

**Comment by csrwng** [comment]
/hold
/lgtm
/approve
Let's hold until we have a working CI

**Pull request merged**